### PR TITLE
Add priority class name, use same temp directory as before

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
Added the `gardener-shoot-control-plane` priority class to the etcd pods and changed the temporary directory path for the snapstore from `/tmp` to `/var/etcd/data/temp`.
Also, changed back the value of `--etcd-connection-timeout` to `5m` (instead of `30s`) as it was before introduction of etcd-druid.

**Release note**:

```improvement operator
Added the `gardener-shoot-control-plane` priority class to the etcd pods and changed the temporary directory path for the snapstore from `/tmp` to `/var/etcd/data/temp`.
```

### CODE REVIEW

File: charts/etcd/templates/etcd-statefulset.yaml [hunk 1]
```diff
@@ -48,6 +48,9 @@ spec:
       - ip: "127.0.0.1"
         hostnames:
         - {{ .Values.name }}-local
+      {{- if .Values.priorityClassName }}
+      priorityClassName: {{ .Values.priorityClassName }}
```
Comment (rfranzke) [inline charts/etcd/templates/etcd-statefulset.yaml:52]: FYI: https://github.com/gardener/gardener/blob/v1.1.6/charts/seed-controlplane/charts/etcd/templates/etcd-statefulset.yaml#L35

File: charts/etcd/values.yaml [hunk 2]
```diff
@@ -31,8 +32,8 @@ etcd:
 
 backup:
   pullPolicy: IfNotPresent
-  snapstoreTempDir: "/tmp"
-  etcdConnectionTimeout: 30
+  snapstoreTempDir: "/var/etcd/data/temp"
+  etcdConnectionTimeout: 5m
```
Comment (rfranzke) [inline charts/etcd/values.yaml:36]: https://github.com/gardener/gardener-extension-provider-aws/blob/v1.4.0/vendor/github.com/gardener/gardener-extensions/pkg/webhook/controlplane/etcd.go#L65

File: charts/etcd/values.yaml [hunk 3]
```diff
@@ -31,8 +32,8 @@ etcd:
 
 backup:
   pullPolicy: IfNotPresent
-  snapstoreTempDir: "/tmp"
-  etcdConnectionTimeout: 30
+  snapstoreTempDir: "/var/etcd/data/temp"
```
Comment (rfranzke) [inline charts/etcd/values.yaml:35]: https://github.com/gardener/gardener-extension-provider-aws/blob/v1.4.0/vendor/github.com/gardener/gardener-extensions/pkg/webhook/controlplane/etcd.go#L69

File: go.mod [hunk 4]
```diff
@@ -35,6 +35,7 @@ require (
 	k8s.io/klog v1.0.0 // indirect
 	k8s.io/utils v0.0.0-20191218082557-f07c713de883 // indirect
 	sigs.k8s.io/controller-runtime v0.4.0
+	sigs.k8s.io/controller-tools v0.2.4 // indirect
```
Comment (swapnilgm) [inline go.mod:38]: Any idea from where this is coming?

File: controllers/etcd_controller.go [hunk 5]
```diff
@@ -808,7 +808,7 @@ func (r *EtcdReconciler) getMapFromEtcd(etcd *druidv1alpha1.Etcd) (map[string]in
 		etcdValues["image"] = etcd.Spec.Etcd.Image
 	}
 
-	var quota int64 = 2 * 1024 * 1024 * 1024 // 2Gib
+	var quota int64 = 8 * 1024 * 1024 * 1024 // 8Gi
```
Comment (swapnilgm) [inline controllers/etcd_controller.go:811]: Suggestion for line below:
```suggestion
	var deltaSnapshotMemoryLimit int64 = 104857600 //100Mib
```

File: go.mod (hunk 4 above)
Comment (rfranzke) [reply go.mod:38]: I ran `make generate`

File: controllers/etcd_controller.go (hunk 5 above)
Comment (rfranzke) [reply controllers/etcd_controller.go:811]: done

**Pull request merged**